src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -155,67 +163,224 @@ private void InitializeDefaultAutoCompleteCandidates()
-            NoRecommendationsOrLowConfidenceTitle = Resources.AutocompleteNoRecommendationsTitle;
-            NoRecommendationsOrLowConfidenceMessage = Resources.AutocompleteNoRecommendationsMessage;
-            */
 
+            // Set port info
+            request.Port.Direction = portInfo.PortType == PortType.Input ? PortType.Input.ToString().ToLower() : PortType.Output.ToString().ToLower();
+            request.Port.Name = portInfo.Name;
+            request.Port.KeepListStructure = portInfo.KeepListStructure.ToString();
+            request.Port.ListAtLevel = portInfo.Level;
 
-            // Case 3: Confidence score are under a threshold, assuming the minimum value is 50 and some result nodes are under it
-            /*
-            FilteredResults = DefaultResults.Where(e => e.Name == "Watch 3D" || e.Name == "Python Script").ToList();
+            // Set host info
+            request.Host.Name = dynamoViewModel.Model.HostAnalyticsInfo.HostName;
+            request.Host.Version = dynamoViewModel.Model.HostVersion;
 
-            foreach (var item in FilteredResults)
+            // Set packages info
+            var packageManager = dynamoViewModel.Model.ExtensionManager.Extensions.OfType<PackageManagerExtension>().FirstOrDefault();
+
+            foreach (var pkg in packageManager.PackageLoader.LocalPackages)
             {
-                item.Model.AutoCompletionNodeMachineLearningInfo.ViewConfidenceScoreRecentUse = true;
-                item.Model.AutoCompletionNodeMachineLearningInfo.IsByUse = false;
+                request.Packages = request.Packages.Append(new PackageItem(pkg.Name, pkg.VersionName));
             }
-            FilteredResults.ToList()[0].Model.AutoCompletionNodeMachineLearningInfo.ConfidenceScore = 50;
-            FilteredResults.ToList()[1].Model.AutoCompletionNodeMachineLearningInfo.ConfidenceScore = 40;
-            RaisePropertyChanged("ViewConfidenceScoreRecentUse");
 
-            DisplayNoRecommendationsLowConfidence = !FilteredResults.Where(n => n.Model.AutoCompletionNodeMachineLearningInfo.ConfidenceScore >= 50).Any();
-            NoRecommendationsOrLowConfidenceMessage = Resources.AutocompleteNoRecommendationsMessage;
-            DisplayLowConfidence = FilteredResults.Where(n => n.Model.AutoCompletionNodeMachineLearningInfo.IsByRecommendation).Count() == FilteredResults.Count();
-            */
+            // Set context info
+            // Considering upstream nodes when the PortType is output and considering downstream nodes when the PortType is input. 
+            var connectors = nodeInfo.AllConnectors;
 
-            // Case 4: Confidence score are under a threshold, assuming the minimum value is 50 and all results nodes are under it
-            /*
-            FilteredResults = DefaultResults.Where(e => e.Name == "Watch 3D" || e.Name == "Python Script").ToList();
-            foreach (var item in FilteredResults)
+            foreach (var connector in connectors)
```

Comment:
If I understand well, this would retrieve the fist level of neighbors of the current node as context, right?
It is not very clear in the schema description, and I may have confused you during our discussion - apologies for this. For context, we expect all nodes (and connections) that can be visited from the node that triggers the query (the whole connected component that includes the query node).

Reply:
Yes currently, it is sending only the first level of neighbors. I will look into adding the remaining information.

Reply:
there should be some code for this floating around for sure.

Reply:
Made changes to the Context property which will have all the information regarding the connected component